daemon/algod/api/server/v2/handlers.go

```diff
@@ -1029,6 +1044,98 @@ func (v2 *Handlers) RawTransactionAsync(ctx echo.Context) error {
 	return ctx.NoContent(http.StatusOK)
 }
 
+// AccountAssetsInformation looks up an account's asset holdings.
+// (GET /v2/accounts/{address}/assets)
+func (v2 *Handlers) AccountAssetsInformation(ctx echo.Context, address string, params model.AccountAssetsInformationParams) error {
+	if !v2.Node.Config().EnableExperimentalAPI {
+		return ctx.String(http.StatusNotFound, "/v2/accounts/{address}/assets was not enabled in the configuration file by setting the EnableExperimentalAPI to true")
+	}
+
+	addr, err := basics.UnmarshalChecksumAddress(address)
+	if err != nil {
+		return badRequest(ctx, err, errFailedToParseAddress, v2.Log)
+	}
+
+	var assetGreaterThan uint64 = 0
+	if params.Next != nil {
+		agt, err0 := strconv.ParseUint(*params.Next, 10, 64)
+		if err0 != nil {
+			return badRequest(ctx, err0, fmt.Sprintf("%s: %v", errUnableToParseNext, err0), v2.Log)
+		}
+		assetGreaterThan = agt
+	}
+
+	if params.Limit != nil {
+		if *params.Limit <= 0 {
+			return badRequest(ctx, errors.New(errInvalidLimit), errInvalidLimit, v2.Log)
+		}
+
+		if *params.Limit > MaxAssetResults {
+			limitErrMsg := fmt.Sprintf("limit %d exceeds max assets single batch limit %d", *params.Limit, MaxAssetResults)
+			return badRequest(ctx, errors.New(limitErrMsg), limitErrMsg, v2.Log)
+		}
+	} else {
+		// default limit
+		l := DefaultAssetResults
+		params.Limit = &l
+	}
+
+	ledger := v2.Node.LedgerForAPI()
+
+	// Logic
+	// 1. Get the account's asset holdings subject to limits
+	// 2. Handle empty response
+	// 3. Prepare JSON response
+	lastRound := ledger.Latest()
+
+	records, lookupRound, err := ledger.LookupAssets(lastRound, addr, basics.AssetIndex(assetGreaterThan), *params.Limit)
```

Comment:
I considered this first, a few complications with this (and why I decided against it).

- Consistency with the indexer API I think is important for end users. The indexer returns the last asset id in the batch as the input token to the next group: https://github.com/algorand/indexer/blob/f82384df31aa628decf24a972f5f79d3ddeb2c71/api/handlers.go#L307
- Resources (assets and apps) are stored in the same `resources` table. The query to do what you suggested thus becomes complicated or more likely becomes multiple queries (since you have to sort by asset id, then find items X through X+10 in a collection inside of SQL).

Reply:
The token based approach also has the downside that changes to assets in early pages (i.e. opting in or out) between requests will shift the results for later pages. Anchoring on asset IDs is stable by comparison